### DISCUSSION TITLE
Extract all the fields listed under the different "category" section in TypeScript files in the form file.

### DISCUSSION THREAD

**Discussion by EdwinThomas11** [serenity-is/Serenity#7060] [Help] [answered]
Hello @VictorTomaili ,

I'm wondering if it's feasible to retrieve all the fields listed under different categories.

For instance:

[Category("Category 1")]
public String Form1 { get; set; }

[Category("Category 2")]
public String Form2 { get; set; }

I'm looking to gather all the fields along with their respective categories. Could you please assist with this? Thank you!

**Comment by VictorTomaili** [comment, marked as answer]
You can see the function `Q.ScriptData.ensure('Form.Administration.User')` or module way `ScriptData.ensure('Form.Administration.User')`

**Reply by EdwinThomas11** [reply]
This is precisely what I was looking for. Appreciate the immediate response!
